src/js/components/BreadcrumbSegment.js

```diff
@@ -0,0 +1,58 @@
+import mixin from 'reactjs-mixin';
+/* eslint-disable no-unused-vars */
+import React, {PropTypes} from 'react';
+/* eslint-enable no-unused-vars */
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import BreadcrumbLink from './BreadcrumbLink';
+
+class BreadcrumbSegment extends mixin(StoreMixin) {
```

Comment:
Nvm, I brought it back, because it's not possible to do `extends mixin(BreadcrumbSegment, StoreMixin) {` at the moment as the first mixin is a react component. Plus I foresee all components that extend `BreadcrumbSegment` to use the storemixin.